### PR TITLE
libfprint-2-tod1-broadcom: init at 5.12.018

### PR DESCRIPTION
**Pull request by pitkling**
## Description of changes

Added a linux driver for broadcom fingerprint readers (used, e.g., in DELL Latitude 7440 laptops) to use with libfprint-tod.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pitkling** [comment]
**Note:** I just saw that there is a PR draft for an older version of this driver, see #246503. As @totoroot mentions there, he hit a roadblock concerning the loading of the binary firmware files which the driver expects in a fixed location (outside `/nix/store`). My PR handles that by redirection `fopen()` calls to the corresponding location, similar to some (very similar to what was done in #260715 in another context).

**Comment by pitkling** [comment]
As indicated by the `nixpkgs-check-by-name` check that was/is failing, I moved the driver to `pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix`. However, the check is still failing, now stating that `This attribute defined by […]/libfprint-2-tod1-broadcom/pacakge.nix is not a derivation`. However, I think it is? Any hints on what I'm doing wrong?

**Comment by pitkling** [comment]
Ok, I fixed the `nixpkgs-check-by-name` check (I misinterpreted the check's output and was not aware that "by-name" packages by default are not required to be added to `pkgs/top-level/all-packages.nix`).

File: maintainers/maintainer-list.nix
Comment (gador) [inline maintainers/maintainer-list.nix:15827]: Please add your maintainer name as a separate commit

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (gador) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:22]:
```suggestion
      url    = "http://dell.archive.canonical.com/updates/pool/public/libf/libfprint-2-tod1-broadcom/libfprint-2-tod1-broadcom_${version}.orig.tar.gz";
```

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (gador) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:23]:
```suggestion
      hash = "sha256-0C2PpYpEJNrU+8NT95w4QV0J5nHQisMY94Czw3jQOzw=";
```

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (gador) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:57]:
```suggestion
    url    = "http://dell.archive.canonical.com/updates/pool/public/libf/libfprint-2-tod1-broadcom/libfprint-2-tod1-broadcom_${version}.orig.tar.gz";
```

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (gador) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:58]:
```suggestion
    hash = "sha256-0C2PpYpEJNrU+8NT95w4QV0J5nHQisMY94Czw3jQOzw=";
```

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (gador) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:66]: do we really need the LICENSE file on the installed derivation?

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (gador) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:21]: If this is the same package (and therefore the same `src`), wouldn't it make sense to just put the whole  `src = ` statement inside the `let` statement (just like `version`) and inherit it for `unpacked`  and `libfprint-2-tod1-broadcom`?

**Review by gador — CHANGES_REQUESTED** [review]

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 326272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libfprint-2-tod1-broadcom</li>
</ul>
</details>

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:63]: If there is no configure script, we don't need to disable the configure phase.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: I guess you're right, I just included it since it was also installed in the package file from some other Linux distribution (don't remember which one). Anyway, I removed it.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: Good point, thanks!

**Comment by gador** [comment]
awesome, lgtm. Could you just squash your commits?

**Comment by pitkling** [comment]
Thanks for the feedback, @gador and @SuperSandro2000. Should all be addressed.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 326272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libfprint-2-tod1-broadcom</li>
</ul>
</details>

**Comment by totoroot** [comment]
@pitkling Thanks for working on this! I tried the changes on your branch, but I'm getting a test failure locally:
```
error: builder for '/nix/store/9gjlwh13fr53x9nsr0zmgg13ifdwxki7-fprintd-tod-1.90.9.drv' failed with exit code 12;
       last 10 log lines:
       > 153/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_script                                                     FAIL            0.44s   exit status 1
       >
       > Ok:                 148
       > Expected Fail:      0
       > Fail:               12
       > Unexpected Pass:    0
       > Skipped:            0
       > Timeout:            0
       >
       > Full log written to /build/source/build/meson-logs/testlog.txt
       For full logs, run 'nix log /nix/store/9gjlwh13fr53x9nsr0zmgg13ifdwxki7-fprintd-tod-1.90.9.drv'.
error: 1 dependencies of derivation '/nix/store/byqsadln65n4v8h38k5sj2mvw1f96kfa-chfn.pam.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i0lg3lhlyn80zh1w0agi636m3zmp11nd-dbus-1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5gnwb40h0fyy2sbwjjlbjn9lpk8sf59h-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0figcyj70qsbli9j656r3b566x1khyja-nixos-system-xxx-24.11.20240712.7e7c39e.drv' failed to build
```

Partial output of `nix log`:
```
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/test_fprintd_utils.py", line 203, in setUp
    self.setup_device()
  File "/build/source/tests/test_fprintd_utils.py", line 90, in setup_device
    self.device_path = self.obj_fprintd_mock.AddDevice(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/01mps72j158sz693xl6sjcixaz6wbld7-python3.12-dbus-python-1.3.2/lib/python3.12/site-packages/dbus/proxies.py", line 141, in __call__
    return self._connection.call_blocking(self._named_service,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/01mps72j158sz693xl6sjcixaz6wbld7-python3.12-dbus-python-1.3.2/lib/python3.12/site-packages/dbus/connection.py", line 634, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.InvalidArgs: Invalid arguments: More items found in D-Bus signature than in Python arguments

----------------------------------------------------------------------
Ran 1 test in 0.288s

FAILED (errors=1)


[154-160/160] 🌔 fprintd:dist / check-translations                                                                                                                                          0/30s^M154/160 fprintd:>
[155-160/160] 🌕 fprintd:dist / check-translations                                                                                                                                          0/30s^M155/160 fprintd:>
[156-160/160] 🌖 fprintd:dist / check-translations                                                                                                                                          0/30s^M156/160 fprintd:>
[157-160/160] 🌗 fprintd:daemon+fprintd+FPrintdVirtualDeviceVerificationTests / FPrintdVirtualDeviceVerificationTests.test_multiple_verify                                                  2/30s^M157/160 fprintd:>
[158-160/160] 🌘 fprintd:daemon+fprintd+FPrintdVirtualDeviceTest / FPrintdVirtualDeviceTest.test_verify_done_disconnect                                                                     2/30s^M158/160 fprintd:>
[159-160/160] 🌑 fprintd:daemon+fprintd+FPrintdVirtualDeviceTest / FPrintdVirtualDeviceTest.test_verify_done_disconnect                                                                     2/30s^M[159-160/160] 🌒>
[160/160] 🌓 fprintd:daemon+fprintd+FPrintdVirtualDeviceTest / FPrintdVirtualDeviceTest.test_verify_done_disconnect                                                                         2/30s^M[160/160] 🌔 fpr>

Summary of Failures:

133/160 fprintd:+test_fprintd_utils+TestFprintdUtils / TestFprintdUtils.test_fprintd_delete                                                                        FAIL            0.47s   exit status 1
135/160 fprintd:+test_fprintd_utils+TestFprintdUtils / TestFprintdUtils.test_fprintd_list                                                                          FAIL            0.45s   exit status 1
136/160 fprintd:+test_fprintd_utils+TestFprintdUtils / TestFprintdUtils.test_fprintd_enroll                                                                        FAIL            0.51s   exit status 1
137/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_list_all_fingers                                                  FAIL            0.42s   exit status 1
143/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_multiple_verify_fails                                             FAIL            0.48s   exit status 1
144/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_any_finger_no_identification                               FAIL            0.41s   exit status 1
146/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_any_finger_identification                                  FAIL            0.47s   exit status 1
147/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify                                                            FAIL            0.56s   exit status 1
148/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_not_enrolled_fingers                                       FAIL            0.40s   exit status 1
150/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_no_enrolled_fingers                                        FAIL            0.51s   exit status 1
151/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_enrolled_fingers                                           FAIL            0.53s   exit status 1
153/160 fprintd:+test_fprintd_utils+TestFprintdUtilsVerify / TestFprintdUtilsVerify.test_fprintd_verify_script                                                     FAIL            0.44s   exit status 1

Ok:                 148
Expected Fail:      0
Fail:               12
Unexpected Pass:    0
Skipped:            0
Timeout:            0

Full log written to /build/source/build/meson-logs/testlog.txt
```

Any idea why this is happening?

**Comment by pitkling** [comment]
> @pitkling Thanks for working on this! I tried the changes on your branch, but I'm getting a test failure locally:
> [..]
> Any idea why this is happening?

Not sure what's the problem. Which NixOS branch are you on? If you're on unstable, maybe it is related to or has some interplay with recent updates to fprintd (see issues #325836 and #326105)? While those are for fprintd and not fprintd-tod, the latter is an overwrite of the former. The fix for those issues (0cecec07191f99c5e7ca43ebf06dad966f09fb61) seems to add a patch. Maybe your NixOS branch does not yet have that patch or that patch has problems with fprintd 1.90 (which seems to be the version used by fprint-tod). Anyway, these are just some quick first guesses, didn't have time to take a closer look yet.

I'm using this on NixOS 24.05 (with some minor, unrelated patches/unstable packages) and didn't see any problems with the fprintd tests.

**Comment by pitkling** [comment]
FYI: The force-push was just a rebase (without any other changes) onto current master; sorry for the unnecessary noise.

**Comment by totoroot** [comment]
I just tried rebuilding my system and it suddenly worked. I'm on unstable, so you might be right that it was the linked issue with `fprintd` and the merged changes have made it to unstable over night.

Anyways, I can confirm that the driver works as intended with the following config: :tada:
```
  services.fprintd = {
    enable = true;
    package = pkgs.fork.fprintd-tod;
    tod = {
      enable = true;
      driver = pkgs.fork.libfprint-2-tod1-broadcom;
    };
  };
```

With `fork` being set to this branch in my flake config like so:
```
nixpkgs-fork.url = "github:pitkling/nixpkgs/libfprint-2-tod1-broadcom";
```

Again, thank you very much for getting this to work! :bouquet:

**Review by totoroot — APPROVED** [review]

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:65]: Good point. I copy-pasted the patchelfUnstable usage from another nixpkgs package and didn't think too much about this, but you're right. I will update it accordingly.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:59]: That path is not the target directory but a path from the unpacked source archive, so it must be hard-coded. But I just realized that my meta attributes don't indicate that the binary driver is only for x86_64 architectures. I'll update the meta information to

```
meta = with lib; {
  …
  platforms = [ "x86_64-linux" ];
  …
};
```
Then this should be fine, right?

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (jtojnar) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:55]: Would be nice if you could use `nixfmt-rfc-style` to format this since it is going to be enforced for `by-name` packages soon.

Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: Done, thanks for the pointer!

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: Done.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:35]: It probably would; to be honest I didn't know about it before and didn't test it.

Still, this would affect the whole process, not only the calls in the library `libfprint-2-tod-1-broadcom.so` into which I inject the wrapper library, would it not? I guess in principle that is okay and should not be a problem in this case. But one could see it as a benefit that the wrapper library affects only those calls that are really necessary, and in particular only those from the corresponding library.

Personally I would probably still opt for this "less invasive" approach (if I understood `libdirect` right and my approach does in fact affect only necessary calls). But I am not too experienced in this respect, so if you think the added complexity of the wrapper-lib is a major drawback, I can try an alternative implementation via `libredirect`.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: Indeed, not at all… I removed the unnecessary `unpacked` derivation and use `src` directly instead, which gets a proper name in the store via the `pname` and `version` attributes that I added.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:57]: I see your point with the hooks, thanks for pointing that out.

Concerning using the default `installPhas` in this case: This might be a stupid question, but what does the default `installPhase` actually do if there is no Makefile (as in this case; the downloaded archive contains only an `install.sh` script for installation)? I was under the impression that, without a Makefile, it does nothing (which is what the [manual seems to indicate](https://nixos.org/manual/nixpkgs/stable/#ssec-install-phase)).

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/wrapper-lib.c
Comment (jtojnar) [inline pkgs/by-name/li/libfprint-2-tod1-broadcom/wrapper-lib.c:17]: Mixed indentation here.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (jtojnar) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:57]: It will not run `make install` but hooks will still be called:

https://github.com/NixOS/nixpkgs/blob/e5bc1cb431307c0a3943615151aeff0b9cc246c3/pkgs/stdenv/generic/setup.sh#L1449-L1452

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (jtojnar) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:27]: I do not really recall how ELF symbol resolution and `--rename-dynamic-symbols` specifically works. You are probably right that this is less invasive than `LD_PRELOAD`.

Ideally, we would modify `libredirect` to allow using it with `--rename-dynamic-symbols` but we do not need to do that here.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (jtojnar) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: I meant more like why is the default `patchelf` not sufficient – for example, if `--rename-dynamic-symbols` is not available on `patchelf`. This will help future maintainers to decide if it is fine to switch to regular `patchelf` when it is eventually updated.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:27]: Just for future reference: The technique on which `--rename-dynamic-symbols` is based on is described [here](https://tapesoftware.net/replace-symbol/) (also referenced in my code). The rough idea is explained under the heading **patchelf --replace-symbol**. Also, I just realized that the page mentions that this approach is less invasive than `LD_PRELOAD` (under the heading **Advantages**).

So, for now I'll leave it with the current implementation via the wrapper-library and mark this conversation as resolved.

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix]: Ah, sure, makes sense. I added a corresponding comment (not in `nativeBuildInput` but  in the module's arguments at the file's top).

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/wrapper-lib.c
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/wrapper-lib.c]: fixed

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:57]: Thanks for the pointer. For now I just added the hooks. But I assume what you actually suggest is to not use `installPhase` at all. Instead of "installing" the files (like the dynamic library) I can directly use those in the unpacked driver archive placed in the nix store. Will try this tomorrow or so.

**Comment by pitkling** [comment]
@gador Sorry for the re-requested review, but Github says I didn't address some of your requested changes although I think I did. Probably because of a recent force-push? I assume I should have not force-pushed my recent changes but squashed them later on…

**Comment by gador** [comment]
> @gador Sorry for the re-requested review, but Github says I didn't address some of your requested changes although I think I did. Probably because of a recent force-push? I assume I should have not force-pushed my recent changes but squashed them later on…

All my concerns are addressed, thanks! I leave it up to @jtojnar to merge this, as I think the discussion with the `installPhase` isn't yet resolved?

**Review by gador — APPROVED** [review]

File: pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix
Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:60]: @jtojnar: After having a closer look, I'm not sure how to

> […] rely on the default `installPhase`. […]

for the main derivation as you suggested. I first thought you meant to directly rely on the path to the dynamic library from the unpacked source, since the wrapper library has a runtime dependency on the unpacked source anyway, but that seems not very clean and the main derivation still needs to install the udev rules in  `$out/lib/udev/rules.d/` for `services.udev.packages` to work correctly, right?

I also checked the other libfprint-2-tod1-* packages (e.g., [libfprint-2-tod1-vfs0090](https://github.com/NixOS/nixpkgs/blob/681e984f8e3d66f4fd8a94ea28bb7a8b87b27216/pkgs/development/libraries/libfprint-2-tod1-vfs0090/default.nix#L24-L31), which use a similar overwrite of the `installPhase` as I do now (with the proper hooks).

So I currently would simply leave the overwritten `installPhase` as is (now with the proper hooks). But let me know if I'm overlooking something.

Comment (jtojnar) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:60]: > […] rely on the default `installPhase`. […]

Sorry, that indeed does not apply here. It is part of my reply template and I forgot to remove the part here this time.

---

Two derivation approach is needed here since we cannot have store path references in fixed-output derivations llike `fetchzip`.

Comment (pitkling) [reply pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix:60]: Ah, no wonder I couldn't figure it out 😅.

Then I'm marking this as resolved. Let me know if there's anything else I should address.

**Comment by pitkling** [comment]
@gador @jtojnar 

Just wanted to ping again, to see if there's anything else I can do or if this can be merged? I think @gador was happy with the current state and at least I think I addressed everything @jtojnar mentioned. But no hurry if anyone of you still needs more time to take a closer look at this.

In fact, I did just push two (in my view) very minor changes. The force push was just to flip the commit order and have the changes in `maintainer-list.nix` as the first commit (as requested in the contribution guidelines). I left the two actual changes as two separate commits, so that it is easier to see what I did; let me know if I should squash them. Quick summary of the two new commits:

* The first one adds a filter in the `src` definition of the `wrapperLib` derivation to avoid unnecessary files in the corresponding nix store path.
* The second commit uses the `name` attribute to (imho) improve the names of the generated store paths slightly.

**Comment by gador** [comment]
>I think @gador was happy with the current state and at least I think I addressed everything @jtojnar mentioned. 

Yes. 

>     * The second commit uses the `name` attribute to (imho) improve the names of the generated store paths slightly.

I would advice against it. There is a PR from 2018 to generalize your idea (of including version attribute in the store name) (https://github.com/NixOS/nixpkgs/pull/49862) and even an RFC (https://github.com/NixOS/rfcs/pull/171) which has some pro and cons (e.g. https://github.com/NixOS/rfcs/pull/171#issuecomment-2082993487) but genereally I would like to see the package namining scheme unchanged for this PR.

This can be changed tree-wide if the RFC ever gets accepted.

**Comment by pitkling** [comment]
> > ```
> > * The second commit uses the `name` attribute to (imho) improve the names of the generated store paths slightly.
> > ```
> 
> […] I would advise against it. There is a PR from 2018 to generalize your idea (of including version attribute in the store name) (#49862) and even an RFC ([NixOS/rfcs#171](https://github.com/NixOS/rfcs/pull/171)) which has some pro and cons (e.g. [NixOS/rfcs#171 (comment)](https://github.com/NixOS/rfcs/pull/171#issuecomment-2082993487)) but genereally I would like to see the package namining scheme unchanged for this PR. […]

Ah, I didn't know this, sorry. Thanks for the quick response! I just added a commit that sticks again to the `pname` and `version` attributes instead of including the `version` manually in `name`. So together, the last two commits now simply remove some unnecessary parenthesis and add `"wrapper-lib"` into the name of the `wrapper-lib`'s `src` attribute.

**Comment by gador** [comment]
@pitkling perfect, thanks. Maybe juist squash the last two commits?

**Comment by pitkling** [comment]
> @pitkling perfect, thanks. Maybe juist squash the last two commits?

Sure, done.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 326272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libfprint-2-tod1-broadcom</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
libfprint-2-tod1-broadcom:
<details>

warning: EvalError
Cannot evaluate attribute ‘libfprint-2-tod1-broadcom’ in ‘/nix/store/n5f84ql9vgyrrlw7xggw5p75q05jv0l5-nixpkgs’.
</details>
</details>

**Pull request merged**
